src/evaluators/BinaryExpression.js

```diff
@@ -170,8 +177,14 @@ export function computeBinary(
 
   if (lval instanceof AbstractValue || rval instanceof AbstractValue) {
     let type = getPureBinaryOperationResultType(realm, op, lval, rval, lloc, rloc);
-    return realm.createAbstract(new TypesDomain(type), ValuesDomain.topVal, [lval, rval], ([lnode, rnode]) =>
-      t.binaryExpression(op, lnode, rnode)
+    let [hash, args] = hashBinary(op, lval, rval);
+    return realm.createAbstract(
+      new TypesDomain(type),
+      ValuesDomain.topVal,
+      hash,
+      args,
+      ([lnode, rnode]) => t.binaryExpression(op, lnode, rnode),
+      op
```

Comment:
Kind is a generic name and the use case is not documented as such. Abstract values that result from expressions will never be builtin (or intrinsic) so there should be no clash.